### PR TITLE
Add flagext package

### PR DESCRIPTION
**Pull request by aknuds1**
Add `flagext` package from Cortex. Will be needed by the `kv` package.

### CODE REVIEW

File: flagext/deprecated.go [hunk 1]
```diff
@@ -1,38 +0,0 @@
-package flagext
```
Comment (pstibrany) [inline flagext/deprecated.go]: Why not keep this as well, perhaps with modified metric name? (It's used to mark some flags as deprecated).

**Review by pstibrany — APPROVED** [review]
Thank you!

File: flagext/deprecated.go (hunk 1 above)
Comment (aknuds1) [reply flagext/deprecated.go]: @pstibrany there was an issue with a dependency on a global logger, and it was the easy fix to remove it. I can put it back though, rewritten with parameterized logger.

Comment (pstibrany) [reply flagext/deprecated.go]: I see. It seems generic enough to be usable for other projects too, but I'm not sure if Tempo/Loki already use it or not. That may guide the decision here.

Comment (aknuds1) [reply flagext/deprecated.go]: @pstibrany I re-added this, with parameterized logger.

**Pull request merged**